src/gordon_gcp/plugins/enricher.py

```diff
@@ -156,28 +156,50 @@ def __init__(self, config, http_client, success_channel, error_channel):
                f'{exception.__class__.__name__}: {exception}')
         raise exceptions.GCPGordonError(msg)
 
-    async def _create_A_rrecord(self, event_data, instance_data, msg_logger):
-        dns_zone = self.config['dns_zone']
-        hostname = event_data['resourceName'].split('/')[-1]
-        access_configs = instance_data['networkInterfaces'][0]['accessConfigs']
-        external_ip = access_configs[0]['natIP']
-
-        name = '.'.join([hostname, dns_zone])
-        msg_logger.debug(f'Creating A record: {name} -> {external_ip}')
+    def _create_A_rrecord(self, fqdn, external_ip, default_ttl, msg_logger):
+        msg_logger.debug(f'Creating A record: {fqdn} -> {external_ip}')
         return {
-            'name': name,
+            'name': fqdn,
             'rrdatas': [external_ip],
             'type': 'A',
-            'ttl': self.config['default_ttl']
+            'ttl': default_ttl
         }
 
-    async def _create_rrecords(self, event_data, msg_logger):
+    async def _get_full_data(self, event_data, msg_logger):
```

Comment:
If we're adding returning an arbitrary dict that we would expect a subclass to use, we should definitely add a docstring describing what it contains.